pkg/auth/oauth2authenticator.go

```diff
@@ -0,0 +1,252 @@
+package auth
+
+import (
+	"context"
+	crypto_rand "crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"html"
+	"math/rand"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/pkg/browser"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CONFIG_KEY_OAUTH_TOKEN string = "OAUTH_TOKEN"
+	OAUTH_CLIENT_ID        string = "0oa37b7oa3zOoDWCe4h7"
```

Comment:
Is this something that maybe should be generated at runtime, or injected at compile time?

Reply:
It's something that is configured/defined by the auth provider, we can inject it during compile time if we like. hardcoding for now was just the most pragmatic way.